plugins/volume/org.xen.xapi.storage.zfs-ng/zfsutils.py

```diff
@@ -22,12 +22,23 @@ def zvol_find_snap_path(dbg, pool_name, snap_id):
             return this_snap_name
     return None
 
-def _zvol_ensure_destroyable(dbg, vol_name):
+def zvol_ensure_destroyable(dbg, vol_name):
     cmd = "zfs list -Hp -o name,origin".split()
-    pattern = vol_name + "@"
+
+    # a zvol depends on a snapshot if its origin is == to that
+    # snapshot, and depends on another zvol if its origin is a
+    # snapshot of that zvol
+    if "@" in vol_name:
```

Comment:
"@" in the name means that it is a snapshot right? So you are speaking about snapshot that depends of another snapshots here?

Reply:
no, this code checks whether other zvols depend on `vol_name`, and the check is different whether  `vol_name` is a snapshot (will appear directly as origin) or a zvol (where we look for snapshots based on it)

Reply:
code disappeared since